--- CMakeDetermineSystem.cmake.org	2026-09-28 20:23:50.044951061 +0200
+++ CMakeDetermineSystem.cmake	2026-09-28 20:24:25.856951048 +0200
@@ -35,7 +35,27 @@
         RESULT_VARIABLE _uname_result)
 
       if(NOT DEFINED CMAKE_SYSTEM_VERSION)
-        set(_ANDROID_API_LEVEL_H $ENV{PREFIX}/include/android/api-level.h)
+
+        if(DEFINED ENV{PREFIX})
+          set(_ANDROID_API_LEVEL_H $ENV{PREFIX}/include/android/api-level.h)
+        else()
+          # Compiler-Pfad prüfen: gcc/g++AmanoTeam-Toolchain-Sysroot
+
+	  message(STATUS " The toolchain used is (CMAKE_TOOLCHAIN_FILE) = ${CMAKE_TOOLCHAIN_FILE}" )
+
+	 if(CMAKE_TOOLCHAIN_FILE MATCHES "android\\-gcc")
+
+            message(STATUS " Using the api-level.h file for GCC")
+
+            set(_ANDROID_API_LEVEL_H /data/local/tmp/sysroot/usr/lib/android-gcc-cross/include/android/api-level.h)
+          else()
+	    message(STATUS "Using the api-level.h file for clang")
+            set(_ANDROID_API_LEVEL_H /data/local/tmp/sysroot/usr/clang19/include/android/api-level.h)
+          endif()
+
+	  message(STATUS "The used api-level.h file is ${_ANDROID_API_LEVEL_H}" )
+        endif()	      
+      
         set(_ANDROID_API_REGEX "#define __ANDROID_API__ ([0-9]+)")
         file(READ ${_ANDROID_API_LEVEL_H} _ANDROID_API_LEVEL_H_CONTENT)
         string(REGEX MATCH ${_ANDROID_API_REGEX} _ANDROID_API_LINE "${_ANDROID_API_LEVEL_H_CONTENT}")

